### PR TITLE
Add anchor links to h1s through h6s

### PR DESCRIPTION
**Pull request by cassidyjames**
Thoughts?

### CODE REVIEW

**Comment by cassidyjames** [comment]
Ah, need to pull in master. One sec.

**Comment by lewisgoddard** [comment]
_Meniacal laughter._

**Comment by danrabbit** [comment]
fixes #580

**Comment by cassidyjames** [comment]
Thank god for Meld.

File: scripts/docs/main.js [hunk 1]
```diff
@@ -19,6 +19,11 @@ $(document).ready(function() {
         hljs.highlightBlock(block);
     });
 
+	// Anchor headings
```
Comment (cassidyjames) [inline scripts/docs/main.js]: Whoopsie, meld put in a tab instead of 4 spaces. Fixing...

**Comment by lewisgoddard** [comment]
I can't hover over the link on `h2`s

**Comment by cassidyjames** [comment]
@lewisgoddard ah, the icon isn't actually hoverable. The whole heading is. Lemme poke at it.

**Comment by lewisgoddard** [comment]
I'd go with what I said on Slack, prepend an `<i class="fa fa-link"></i>` on the `h#` and just hide it with CSS when not hovered.

**Comment by danrabbit** [comment]
- [ ] Sidebar is busted. No subheadings

**Comment by lewisgoddard** [comment]
You've broken the next page links a little too. Maybe we should add a class to those that can be targeted with `:not('.next-page')`

**Comment by cassidyjames** [comment]
@lewisgoddard hm, imho the next page links should not be `<h4>`s. They're not headings.

**Comment by lewisgoddard** [comment]
I think that's what they were previously, and I can't add classes or ids to paragraphs.

**Comment by danrabbit** [comment]
Seems to work as expected for me. Link can be copied. Open in new tab works. Smooth scrolling still works.

**Pull request merged**